api-tests/features/return-codes.feature

```diff
@@ -0,0 +1,174 @@
+      When I submit an 'appTriage' event
+      Then I get a 'dcmaw' CRI response
+      When I submit 'kenneth-passport-valid' details to the CRI stub
+      Then I get a 'page-dcmaw-success' page response
+      When I submit a 'next' event
+      Then I get an 'address' CRI response
+      When I submit 'kenneth-current' details to the CRI stub
+      Then I get a 'fraud' CRI response
+      When I submit 'kenneth-score-2' details to the CRI stub
+      Then I get a 'page-ipv-success' page response
+      When I submit a 'next' event
+      Then I get an OAuth response
+      When I use the OAuth response to get my identity
+      Then I get a 'P2' identity
+      And I don't get any return codes
+
+  Rule: non-ci-breaching code returned
+    Scenario: Failed identity journey with no CI - user doesn't hold appropriate documents
+      Given I start a new 'medium-confidence' journey
+      Then I get a 'page-ipv-identity-document-start' page response
+      When I submit an 'end' event
+      Then I get a 'page-ipv-identity-postoffice-start' page response
+      When I submit an 'end' event
+      Then I get a 'pyi-escape' page response
+      When I submit a 'end' event
+      Then I get an OAuth response
+      When I use the OAuth response to get my identity
+      Then I get a 'P0' identity
+      And I get 'non-ci-breaching' return code
+
+    Rule: Failed to complete journey
+      Background: Start alternate doc journey with no CI
+        Given I start a new 'medium-confidence' journey
+        Then I get a 'page-ipv-identity-document-start' page response
+        When I submit an 'appTriage' event
+        Then I get a 'dcmaw' CRI response
+        When I get an 'access_denied' OAuth error from the CRI stub
+        Then I get a 'page-multiple-doc-check' page response
+        When I submit a 'drivingLicence' event
+        Then I get a 'drivingLicence' CRI response
+        When I submit 'kenneth-driving-permit-valid' details to the CRI stub
+        Then I get an 'address' CRI response
+        When I submit 'kenneth-current' details to the CRI stub
+        Then I get a 'fraud' CRI response
+        When I submit 'kenneth-score-2' details to the CRI stub
+        Then I get a 'page-pre-experian-kbv-transition' page response
+        When I submit a 'next' event
+        Then I get a 'kbv' CRI response
+        When I submit 'kenneth-thin-file' details to the CRI stub
+        Then I get a 'pyi-cri-escape' page response
+
+      Scenario: Error from f2f CRI
+        When I submit a 'f2f' event
+        Then I get a 'f2f' CRI response
+        When I get an 'access_denied' OAuth error from the CRI stub
+        Then I get a 'pyi-another-way' page response
+        When I submit a 'next' event
+        Then I get an OAuth response
+        When I use the OAuth response to get my identity
+        Then I get a 'P0' identity
+        And I get 'non-ci-breaching' return code
+
+      Scenario: Error from DCMAW CRI
```

Comment:
Not sure whether it's that much more useful to have both of these scenarios?

Reply:
I'd be happy for these to be combined